src/databricks/labs/dqx/llm/llm_pk_detector.py

```diff
@@ -0,0 +1,698 @@
+"""Primary Key Detection using DSPy with Databricks Model Serving."""
+
+import logging
+from typing import Any
+
+import dspy  # type: ignore
+from databricks_langchain import ChatDatabricks  # type: ignore
+from langchain_core.messages import HumanMessage  # type: ignore
+from pyspark.sql import SparkSession
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatabricksLM(dspy.LM):
+    """Custom DSPy LM adapter for Databricks Model Serving."""
+
+    def __init__(self, endpoint: str, temperature: float = 0.1, max_tokens: int = 1000, chat_databricks_cls=None):
+        self.endpoint = endpoint
+        self.temperature = temperature
+        self.max_tokens = max_tokens
+
+        # Allow injection of ChatDatabricks class for testing
+        chat_cls = chat_databricks_cls or ChatDatabricks
+        self.llm = chat_cls(endpoint=endpoint, temperature=temperature, max_tokens=max_tokens)
+        super().__init__(model=endpoint)
+
+    def __call__(self, prompt=None, messages=None, **kwargs):
+        """Call the Databricks model serving endpoint."""
+        try:
+            if messages:
+                response = self.llm.invoke(messages)
+            else:
+                response = self.llm.invoke([HumanMessage(content=prompt)])
+
+            return [response.content]
+
+        except (ConnectionError, TimeoutError, ValueError) as e:
+            print(f"Error calling Databricks model: {e}")
+            return [f"Error: {str(e)}"]
+        except (AttributeError, TypeError, RuntimeError) as e:
+            print(f"Unexpected error calling Databricks model: {e}")
+            return [f"Unexpected error: {str(e)}"]
+
+
+class SparkManager:
```

Comment:
```suggestion
class TableManager:
```
it has little to do with spark, but everything with Table